packages/amplify-cli/src/extensions/amplify-helpers/update-amplify-meta.ts

```diff
@@ -77,13 +77,13 @@ const moveBackendResourcesToCurrentCloudBackend = (resources: $TSObject[]): void
     }
 
     fs.ensureDirSync(targetDir);
+    const isLambdaOrCustom =
+      resource?.service === ServiceName.LambdaFunction || (resource?.service && resource?.service.includes('custom'));
 
     // in the case that the resource is being deleted, the sourceDir won't exist
     if (fs.pathExistsSync(sourceDir)) {
-      fs.copySync(sourceDir, targetDir);
-      if (resource?.service === ServiceName.LambdaFunction || (resource?.service && resource?.service.includes('custom'))) {
-        removeNodeModulesDir(targetDir);
-      }
+      const nodeModulesFilterFn = (src: string): boolean => path.basename(src) !== 'node_modules';
+      fs.copySync(sourceDir, targetDir, { ...(isLambdaOrCustom ? { filter: nodeModulesFilterFn } : {}) });
     }
```

Comment:
The changes look good, if the `removeNodeModulesDir` method is no longer used, can you remove it as well?